app/src/main/java/ch/epfl/sdp/game/PredatorActivity.kt

```diff
@@ -0,0 +1,244 @@
+import ch.epfl.sdp.game.data.Prey
+
+
+class PredatorActivity : AppCompatActivity(), OnTargetSelectedListener {
+    private lateinit var mLocationManager: LocationManager
+    private var _binding: ActivityPredatorBinding? = null
+    private val binding get() = _binding!!
+
+    private var gameID = -1
+    private var playerID = -1
+    private var targetID = -1
+
+    private val lastKnownLocation: Location = Location(0.0, 0.0)
+    private var players = HashMap<Int, Player>()
+    private var preys = HashMap<String, Int>()
+
+    private lateinit var gameTimerFragment: GameTimerFragment
+    private lateinit var targetSelectionFragment: TargetSelectionFragment
+    private lateinit var targetDistanceFragment: TargetDistanceFragment
+    private lateinit var preyFragment: PreyFragment
+
+    private val mLocationListener: LocationListener = object : LocationListener {
+        override fun onLocationChanged(location: android.location.Location) {
+            lastKnownLocation.latitude = location.latitude
+            lastKnownLocation.longitude = location.longitude
+
+            locationSynchronizer.updateOwnLocation(lastKnownLocation)
+
+            if (targetID != TargetSelectionFragment.NO_TARGET) {
+                targetDistanceFragment.distance =
+                        players[targetID]?.lastKnownLocation?.distanceTo(lastKnownLocation)?.toInt() ?: TargetDistanceFragment.NO_DISTANCE
+            }
+        }
+
+        override fun onStatusChanged(provider: String, status: Int, extras: Bundle) {
+            Toast.makeText(applicationContext, "Location: onStatusChanged: $status", Toast.LENGTH_LONG).show()
+        }
+
+        override fun onProviderEnabled(provider: String) {
+            if (targetID != TargetSelectionFragment.NO_TARGET) {
+                targetDistanceFragment.distance = TargetDistanceFragment.NO_DISTANCE
+            }
+        }
+
+        override fun onProviderDisabled(provider: String) {
+            targetDistanceFragment.distance = TargetDistanceFragment.DISABLED
+        }
+    }
+
+    private lateinit var locationSynchronizer: LocationSynchronizer
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        _binding = ActivityPredatorBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Get game information
+        if (savedInstanceState == null) { // First load
+            if(!loadIntentPayload(intent)) {
+                finish()
+                return
+            }
+        }
+
+        loadFragments()
+
+        locationSynchronizer = SimpleLocationSynchronizer(gameID, playerID, MQTTRealTimePubSub(this))
+        locationSynchronizer.setPlayerUpdateListener(object : LocationSynchronizer.PlayerUpdateListener {
+            override fun onPlayerLocationUpdate(playerID: Int, location: Location) {
+                if (players.containsKey(playerID)) {
+                    players[playerID]!!.lastKnownLocation = location
```

Comment:
Are you sure that it won't be null ?

Reply:
Yes, because of the check above. But the array is mutable. Should I make it a val ?
This would fix this issue but could cause problem later if we choose to implement a "leave the game", kick, ban, etc. option of some sort

Reply:
Ok it's fine then